app/models/change_date.rb

```diff
@@ -0,0 +1,49 @@
+class ChangeDate
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveRecord::AttributeAssignment
+
+  attribute :import_date, :date
+
+  def initialize(attributes = {})
+    # We need to do validations before attribute coersion for multiparameter assignment
+    # otherwise we get a MultiparameterAssignmentErrors error propagated when we fail to cast and do not attach the correct error messages for the form.
+    valid_attributes?(attributes)
+
+    super(attributes)
+  end
+
+  def day
+    import_date&.day
+  end
+
+  def month
+    import_date&.month
+  end
+
+  def year
+    import_date&.year
+  end
```

Comment:
maybe you want to delegate these 3 methods to  import date using allow_nil?

Reply:
Good idea :)